app/tests/reader_studies_tests/test_forms.py

```diff
@@ -548,312 +558,446 @@ def test_image_port_only_with_bounding_box(
         reverse_kwargs={"slug": rs.slug},
-        data={"title": "2", "copy_questions": True},
-        user=editor,
+        data={"title": "copy"},
+        user=admin,
         follow=True,
     )
 
     assert response.status_code == 200
-    assert ReaderStudy.objects.count() == 3
-
-    _rs = ReaderStudy.objects.order_by("created").last()
-    assert _rs.title == "2"
-    assert _rs.questions.count() == 2
-    assert _rs.display_sets.count() == 0
-    assert _rs.view_content == {}
-    assert _rs.hanging_protocol is None
-    assert _rs.case_text == {}
-    assert _rs.readers_group.user_set.count() == 0
-    assert _rs.editors_group.user_set.count() == 1
+    assert ReaderStudy.objects.count() == 2
 
-    question.refresh_from_db()
-    copied_question = _rs.questions.first()
+    copied_rs = ReaderStudy.objects.order_by("created").last()
+    assert copied_rs.title == "copy"
 
-    assert question.reader_study == rs
-    assert copied_question.pk != question.pk
-    assert copied_question.reader_study != question.reader_study
+    for field_name in reader_study_copy_fields:
+        field = ReaderStudy._meta.get_field(field_name)
+        original_value, value_in_copy = get_values(rs, copied_rs, field)
+        assert_value_copied(field, original_value, value_in_copy)
 
-    # check that all but the specified fields have been copied over
-    non_copied_fields = [
-        "questionuserobjectpermission",
-        "questiongroupobjectpermission",
-        "answer",
-        "reader_study",
-        "id",
-        "created",
-        "modified",
-        "options",
-    ]
-    for field in copied_question._meta.get_fields():
-        if field.name not in non_copied_fields:
-            assert getattr(copied_question, field.name) == getattr(
-                question, field.name
-            )
+    for field_name in reader_study_non_copy_fields:
+        if (
+            field_name == "readerstudyuserobjectpermission"
+            and len(ReaderStudyUserObjectPermission.allowed_permissions) == 0
+        ):
+            # cannot test if there can be no value
+            continue
 
-    assert copied_question.options.get().title == question.options.get().title
-    assert copied_question.options.get().pk != question.options.get().pk
+        field = ReaderStudy._meta.get_field(field_name)
+        original_value, value_in_copy = get_values(rs, copied_rs, field)
 
-    with django_capture_on_commit_callbacks(execute=True):
-        response = get_view_for_user(
-            viewname="reader-studies:copy",
-            client=client,
-            method=client.post,
-            reverse_kwargs={"slug": rs.slug},
-            data={"title": "3", "copy_display_sets": True},
-            user=editor,
-            follow=True,
-        )
+        if field_name == "readerstudygroupobjectpermission":
+            assert value_in_copy.count() == 10
+            assert set(value_in_copy.all()) != set(original_value.all())
```

Comment:
This tests are testing that the sets are not the same, but shouldn't they be disjoint? This happens in a couple of places.

Generally, it's best to avoid for loops and if statements in tests. If possible, it might be worth trying to rework the tests to use parametrise, and then write different tests for the special cases that require an if statement, putting any common code into a fixture (although we do not worry so much about duplication in tests).